test/Microsoft.AspNet.Mvc.ModelBinding.Test/BodyBindingAndValidationIntegrationTest.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Test
+{
+    public class BodyValidationIntegrationTests
+    {
+        private class Person
+        {
+            [FromBody]
+            [Required]
+            public Address Address { get; set; }
+        }
+
+        private class Address
+        {
+            public string Street { get; set; }
+        }
+
+        [Fact]
+        public async Task BodyBoundOnProperty_RequiredOnProperty_AddsModelStateError()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderModelName = "CustomParameter",
+                },
+                ParameterType = typeof(Person)
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext();
+            var httpContext = operationContext.HttpContext;
+            ConfigureHttpRequest(httpContext.Request, string.Empty);
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var model = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.Equal("The Address field is required.", modelState[""].Errors.Single().ErrorMessage);
+        }
+
+        private class Person4
```

Comment:
Ah, I get it. you're defining a new model class for each test so they are independent.